projects/addon-charts/components/arc-chart/arc-chart.component.ts

```diff
@@ -0,0 +1,107 @@
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    HostBinding,
+    Inject,
+    Input,
+} from '@angular/core';
+import {DomSanitizer, SafeValue} from '@angular/platform-browser';
+import {tuiDefaultProp} from '@taiga-ui/cdk';
+import {TuiSizeXL} from '@taiga-ui/core';
+
+// 3/4 with 1% safety offset
+const ARC = 0.76;
+
+const SIZE: Record<TuiSizeXL, number> = {
+    m: 9,
+    l: 11,
+    xl: 16,
+};
+
+const WIDTH: Record<TuiSizeXL, number> = {
+    m: 0.25,
+    l: 0.375,
+    xl: 0.5625,
+};
+
+const GAP: Record<TuiSizeXL, number> = {
+    m: 0.125,
+    l: 0.1875,
+    xl: 0.25,
+};
+
+@Component({
+    selector: 'tui-arc-chart',
+    templateUrl: './arc-chart.template.html',
+    styleUrls: ['./arc-chart.style.less'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TuiArcChartComponent {
+    @Input()
+    @tuiDefaultProp()
+    value: readonly number[] = [];
+
+    @Input()
+    @HostBinding('attr.data-size')
+    @tuiDefaultProp()
+    size: TuiSizeXL = 'm';
+
+    @Input()
+    @tuiDefaultProp()
+    max = 100;
+
+    @Input()
+    @tuiDefaultProp()
+    minLabel = '0%';
+
+    @Input()
+    @tuiDefaultProp()
+    maxLabel = '100%';
+
+    initialized = false;
+
+    constructor(
+        @Inject(DomSanitizer) private readonly sanitizer: DomSanitizer,
+        @Inject(ChangeDetectorRef) changeDetectorRef: ChangeDetectorRef,
+    ) {
+        // So initial animation works
+        setTimeout(() => {
+            this.initialized = true;
+            changeDetectorRef.markForCheck();
+        });
+    }
+
+    @HostBinding('style.width.rem')
+    @HostBinding('style.height.rem')
+    get width(): number {
+        return SIZE[this.size];
+    }
+
+    @HostBinding('style.strokeWidth.rem')
+    get strokeWidth(): number {
+        return WIDTH[this.size];
+    }
+
+    getInset(index: number): number {
+        return this.strokeWidth / 2 + index * (this.strokeWidth + GAP[this.size]);
+    }
+
+    getDiameter(index: number): number {
+        return SIZE[this.size] - 2 * this.getInset(index);
+    }
+
+    getLength(index: number): number {
+        return Math.PI * this.getDiameter(index) * ARC;
+    }
+
+    getOffset(index: number): number {
+        return this.getLength(index) * (1 - this.value[index] / this.max);
+    }
+
+    getColor(index: number): SafeValue {
+        return this.sanitizer.bypassSecurityTrustStyle(
+            `var(--tui-chart-${index}, var(--tui-support-0${index + 1}))`,
+        );
+    }
```

Comment:
why not move the functionality to ng pipes for prevent unnecessary recalculations?

Reply:
Because these are very cheap and it's more readable this way.